# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/streamconv/nsIStreamConverter.idl
# Commit: c58710d55a77
# Full Hash: c58710d55a771f83a36c1c7504d2edbeb6c1d1b5
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2024-03-20 04:48:34
# Regressor Bug: 1356686
# File Overlap Count: 1
# Description:
#   Bug 1356686 - Do decompression off main thread in content r=necko-reviewers,valentin,extension-reviewers,robwu
#   
#   Content processes will now always retarget delivery of OnDataAvailable for Http
#   channels off the main thread. Consumers that were previously redirecting
#   off-main thread are not affected and their retargeting will stick, but any
# ==============================================================================

diff -r 969c8ce344d6 -r c58710d55a77 netwerk/streamconv/nsIStreamConverter.idl
--- a/netwerk/streamconv/nsIStreamConverter.idl	Tue Mar 19 17:27:22 2024 +0000
+++ b/netwerk/streamconv/nsIStreamConverter.idl	Tue Mar 19 17:27:22 2024 +0000
@@ -96,6 +96,12 @@
                           in nsISupports aCtxt);
 
     /**
+     * This is called after the request has installed the stream converter as its listener
+     * giving the stream converter the option to request retargetting of onDataAvailable.
+     */
+    void maybeRetarget(in nsIRequest request);
+
+    /**
      * Returns the content type that the stream listener passed to asyncConvertData will
      * see on the channel if the conversion is being done from aFromType to * /*.
      *